diff --git a/include/boost/te.hpp b/include/boost/te.hpp
index d35f0fa..6770c86 100644
--- a/include/boost/te.hpp
+++ b/include/boost/te.hpp
@@ -418,6 +418,13 @@ class static_vtable {
 
 namespace detail {
 struct poly_base {
+  poly_base() = default;
+  poly_base(poly_base const&) = default;
+  poly_base(poly_base&&) noexcept = default;
+  poly_base& operator=(poly_base const&) = default;
+  poly_base& operator=(poly_base&&) noexcept = default;
+  virtual ~poly_base() = default;
+
   void** vptr = nullptr;
   virtual void* ptr() const noexcept = 0;
 };
